packages/backend/src/database/migrations/20250404110407_nested_spaces.ts

```diff
@@ -0,0 +1,48 @@
+import { Knex } from 'knex';
+
+const SpacesTableName = 'spaces';
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.schema.alterTable(SpacesTableName, (table) => {
+        table.uuid('parent_space_uuid').nullable();
+        // Enables much faster queries for finding all descendants or ancestors
+        // Examples: https://www.postgresql.org/docs/current/ltree.html#LTREE-EXAMPLE
+        table.specificType('path', 'ltree').nullable();
```

Comment:
are we ok with `path` being `nullable` here even though it's populated in line 29?

I can't use `defaultTo` and reference another column's value (I wanted to use `slug`)

Reply:
what's the downside of making it not nullable?

Reply:
> I can't use defaultTo and reference another column's value (I wanted to use slug)

ah, this makes more sense now! What if you set it as an empty string initially? That way DB constraint is going to be correct and the correct value will be set anyway on line 29